deploy.md

```diff
@@ -161,6 +161,89 @@ services:
 
 `memory` configures a limit or reservation on the amount of memory a container can allocate, set as a string expressing a [byte value](spec.md#specifying-byte-values).
 
+#### devices
+
+`devices` configures reservations of the devices a container can use. It contains a list of reservations, each set as an object with the following parameters: `capabilities`, `driver`, `count`, `device_ids` and `options`.
+
+Devices are reserved using a list of capabilities, making `capabilities` the only required field. A device MUST satisfy all the requested capabilities for a successful reservation.
+
+##### capabilities
+
+`capabilities` are set as a list of strings, expressing both generic and driver specific capabilities.
+The following generic capabilities are recognized today:
+
+- `gpu`: Graphics accelerator
+- `tpu`: AI accelerator
+
+To avoid name clashes, driver specific capabilities MUST be prefixed with the driver name.
+For example, reserving an nVidia CUDA-enabled accelerator might look like this:
+
+```yml
+deploy:
+  resources:
+    reservations:
+      devices:
+        - capabilities: ["nvidia-compute"]
+```
+
+##### driver
+
+A different driver for the reserved device(s) can be requested using `driver` field. The value is specified as a string.
+
+```yml
+deploy:
+  resources:
+    reservations:
+      devices:
+        - capabilities: ["nvidia-compute"]
+          driver: nvidia
+```
+
+##### count
+
+If `count` is set to `-1` or not specified, Compose implementations MUST reserve all devices that satisfy the requested capabilities. Otherwise, Compose implementations MUST reserve at least the number of devices specified. The value is specified as an integer.
```

Comment:
A bit of a nit but I'm wondering if there's a better way to say "all" than `-1`. The only other place that I can think of in the spec where we do something analogous to "all" is for [deploying global services](https://docs.docker.com/compose/compose-file/#deploy). There we do something like this:

```yaml
services:
  worker:
    image: dockersamples/examplevotingapp_worker
    deploy:
      mode: global
```

Maybe we could do something similar by adding a `mode` field that can have values `partial` or `all` with partial being the default:
```yaml
deploy:
  resources:
    reservations:
      devices:
        - capabilities: ["tpu"]
          mode: all
```

Reply:
I agree `-1` is a common technical way to define an "unset" value, but is non-intuitive for a user-friendly file format. docker CLI uses "number or `all`" for `--gpu` flag. We could do the same with the `count` attribute.
Another option is indeed to have a separate attribute (which would be exclusive to `count` definition) to select all devices.

Reply:
I’m leaning towards `count: all`. This is because I’m cautious about the cognitive load of having 3 mutually exclusive ways of requesting a device 🙈

Reply:
Updated - how does this look @chris-crone?

Reply:
LGTM!